### PR TITLE
Move test_approx_eq_modphase into Base.Test

### PR DESCRIPTION
**Pull request by jiahao**
This PR provides a new test function for testing the approximate equality of two vectors (or columns of two matrices) up to a global sign difference. The global sign difference occasionally shows up when testing equality between eigenvectors or singular vectors. These vectors, when normalized, are uniquely defined only up to a global phase (or global sign for the real case). The global phase is usually defined consistently, but may occasionally change due to small differences in floating point rounding noise or rounding modes, or through the use of different conventions. As a result, most tests checking such vectors have to detect and discard such overall phase differences.

`test_approx_eq_vecs` was originally defined within the `linalg` tests as an internal function, but I've since seen situations in other packages (like MultivariateStats.jl) where the same test is required. So it seems useful to have in `Base.Test`. I've left it unexported, as it feels like an expert function of limited use.

### CODE REVIEW

**Comment by timholy** [comment]
I've wanted that functionality myself several times, so I for one appreciate the idea of making it more accessible. However, I'm not sure the name does justice to what the function does: I would expect it to test approximate equality for vectors, not approximate equivalence under global phase. Can it be renamed `test_approx_equiv_vecs_phase` or something?

**Comment by jiahao** [comment]
`test_approx_equiv_vecs_phase` seems rather long. How about `test_eigenvectors_properly` :)

**Comment by timholy** [comment]
I can go with that. I might replace `properly` with `equivalence`, but I'm sure that's just me :smile:.

**Comment by jiahao** [comment]
How about `test_evec_equiv` then?

File: base/test.jl [hunk 1]
```diff
@@ -168,4 +168,20 @@ macro inferred(ex)
     end
 end
 
+#Test equivalence of eigenvectors/singular vectors taking into account possible
+#phase (sign) differences
+function test_approx_eq_vecs{S<:Real,T<:Real}(a::StridedVecOrMat{S},
+    b::StridedVecOrMat{T}, error=nothing)
```
Comment (nalimilan) [inline base/test.jl]: Weird identation here.

File: base/test.jl [hunk 2]
```diff
@@ -168,4 +168,20 @@ macro inferred(ex)
     end
 end
 
+#Test equivalence of eigenvectors/singular vectors taking into account possible
```
Comment (nalimilan) [inline base/test.jl]: "Taking into account" is not very explicit. How about telling what you said in the PR? Anyways, this should go into the docs, not here.

**Comment by nalimilan** [comment]
I agree this can be useful. How about `test_approx_eq_sign/phase`? The fact that it operates on vectors does not need to be in the name (for scalars, this definition does not make much sense, or is equivalent to `test_approx_eq`).

**Comment by toivoh** [comment]
How about `test_approx_proportional`? Isn't that what it does/should do? Or is it undesirable to let the test pass if the vectors don't have approximately the same norms?

**Comment by timholy** [comment]
+1

File: base/test.jl (hunk 2 above)
Comment (jiahao) [inline base/test.jl]: I don't think we provide documentation for unexported functions. Is this useful enough to be exported?

**Comment by jiahao** [comment]
@toivoh this test does not rescale the vectors. I have not found a need to account for renormalizing vectors in practice.

**Comment by jiahao** [comment]
I think I'll go with `test_approx_eq_modphase`. I agree with @nalimilan that the function itself need not contain 'vectors' in the name; however, the fact that the columns of the inputs are being treated as elements of a vector space is critical to defining the function.

**Comment by timholy** [comment]
Sounds great to me.

File: base/test.jl (hunk 2 above)
Comment (nalimilan) [inline base/test.jl]: Ah, since you said you had seen it in other packages, I assumed it would be exported. `@test_approx_eq` is documented at least.

**Comment by ViralBShah** [comment]
Does this need documentation?

It looks like the names in `Tests` could use some bikeshedding.

**Pull request merged**